File: backend/classe/gare.py
class Gare:
    def __init__(self, name,ligne, x, y):
        self.name = name
        self.ligne = ligne
        self.cord = [x, y]
        self.voie=[]
        ligne.add_Gare(self)



    def add_Voie(self,New_Voie):
        self.voie.append(New_Voie)



    def __str__(self):
        return f"Gare(name={self.name}, ligne={self.ligne.name}, nb de voie={len(self.voie)}, coord=({self.cord[0]}, {self.cord[1]}))"

    def get_attr(self, attr_name):
        return getattr(self, attr_name, None)

    def find_voie(self,Gare2):
        for voie in self.voie:
            if voie.get_other(self) == Gare2:
                return voie
        return None

File: backend/src/Gare.java
import java.util.ArrayList;
import java.util.List;

public class Gare {
    private String name;
    private Line ligne;
    private int[] cord;
    private List<Voie> voie;

    public Gare(String name, Line ligne, int x, int y) {
        this.name = name;
        this.ligne = ligne;
        this.cord = new int[]{x, y};
        this.voie = new ArrayList<>();
        ligne.addGare(this);
    }

    public void addVoie(Voie newVoie) {
        voie.add(newVoie);
    }

    @Override
    public String toString() {
        return String.format("Gare(name=%s, ligne=%s, nb de voie=%d, coord=(%d, %d))",
                name, ligne.getName(), voie.size(), cord[0], cord[1]);
    }

    public List<Voie> getVoie() {
        return voie;
    }


    public Voie findVoie(Gare gare2) {
        for (Voie v : voie) {
            if (v.getOther(this) == gare2) {
                return v;
            }
        }
        return null;
    }

    public String getName() {
        return name;
    }
}
